jdaviz/configs/cubeviz/plugins/parsers.py

```diff
@@ -147,63 +285,51 @@ def _parse_jwst_s3d(app, hdulist, data_label, ext='SCI', viewer_name='flux-viewe
                     hdulist[ext].header['MJD-OBS'] = t.mjd
                     break
 
-    if ext == 'DQ':  # DQ flags have no unit
-        flux = hdulist[ext].data << u.dimensionless_unscaled
-    else:
-        unit = u.Unit(hdulist[ext].header.get('BUNIT', 'count'))
-        flux = hdulist[ext].data << unit
-    wcs = WCS(hdulist['SCI'].header, hdulist)  # Everything uses SCI WCS
-    data = Spectrum1D(flux, wcs=wcs)
-
-    # NOTE: Tried to only pass in sliced WCS but got error in Glue.
-    # sliced_wcs = wcs[:, 0, 0]  # Only want wavelengths
-    # data = Spectrum1D(flux, wcs=sliced_wcs)
-
-    app.add_data(data, data_label)
-    app.add_data_to_viewer(viewer_name, data_label)
-    if viewer_name == 'flux-viewer':
-        app.add_data_to_viewer('spectrum-viewer', data_label)
-
 
-def _parse_spectrum1d_3d(app, file_obj):
-    # Load spectrum1d as a cube
+def _parse_spectrum1d_3d(app, file_obj, base_data_label):
+    """Load Spectrum1D as a cube."""
 
-    for attr in ["flux", "mask", "uncertainty"]:
+    for attr in ("flux", "uncertainty", "mask"):
         val = getattr(file_obj, attr)
-        if val is None:
+        if val is None:  # pragma: no cover
             continue
 
-        if attr == "mask":
-            flux = val << file_obj.flux.unit
-        elif attr == "uncertainty":
-            if hasattr(val, "array"):
-                flux = u.Quantity(val.array, file_obj.flux.unit)
-            else:
-                continue
-        else:
+        if attr == "flux":
+            data_type = 'flux'
             flux = val
-
-        flux = np.moveaxis(flux, 1, 0)
+        elif attr == "uncertainty" and hasattr(val, "array"):
+            data_type = 'uncert'
+            flux = val.array << file_obj.flux.unit
+        elif attr == "mask":
+            data_type = 'mask'
+            flux = val << u.dimensionless_unscaled
```

Comment:
Mask should not have units. It is basically data quality flags (unitless). Not sure why it was inheriting science data units.